src/cmd/flux-mini.py

```diff
@@ -32,6 +33,45 @@
 from flux.progress import ProgressBar
 
 
+class Dependency:
+    """Convenience class for handling dependencies
+
+    Splits a dependency URI into fields and returns an RFC 26 dependency
+    entry via the entry attribute.
+    """
+
+    def __init__(self, uri):
+        # replace first ':' with ':FXX' to work around urlparse refusal
+        # to treat integer only path as a scheme:path.
+        uri_string = ":FXX".join(uri.split(":", 1))
```

Comment:
Rather than `"new".join(str.split("old", 1))`, I think you can use `str.replace("old", "new", 1)`:

```python
Python 3.9.5 (default, May 12 2021, 15:26:36)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> uri = "jobid:1"
>>> uri.replace(":", ":FXX", 1)
'jobid:FXX1'
```

Reply:
ooh, nice! Thanks!